sdk/typescript/experimental/src/module.ts

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { SuiObjectRef, CallArg, TypeTag, MoveCallTx } from "./types";
+
+/**
+ * Base class for helping implement MoveCall functionality.
+ * Wraps the module package reference and a name and makes sure transactions
+ * follow the format required by the gateway.
+ *
+ * @example
+ * class DevNetNFT extends sui.MoveModule {
+ *   constructor(public pkg: SuiObjectRef) { super(pkg, 'devnet_nft'); }
+ *
+ *   // Move function is:
+ *   // public entry fun mint(name: vector<u8>, description: vector<u8>, url: vector<u8>)
+ *   public mint(name: string, description: string, url: string): MoveCallTx {
+ *     return this.call('mint', [], [
+ *       { Pure: bcs.ser(bcs.STRING, name).toBytes() },
+ *       { Pure: bcs.ser(bcs.STRING, description).toBytes() },
+ *       { Pure: bcs.ser(bcs.STRING, url).toBytes() }
```

Comment:
The library can figure out whether to use `Pure` or `ObjectArg`, and `String` v.s `Number` by downloading the package object right?

Reply:
We could try to parse function arguments, correct, if we had access to the decompiled sources.